light-client/src/lib.rs

```diff
@@ -121,6 +122,47 @@ impl LightClient for RemoteLightClient {
     }
 }
 
+#[derive(Clone)]
+pub struct UdpLightClient {
+    // url: String,
+    socket: Arc<net::UdpSocket>,
+}
+
+impl UdpLightClient {
+    pub async fn new(url: &str) -> Self {
+        let url = url.replace("udp://", "");
+        let socket = Arc::new(UdpSocket::bind("0.0.0.0:0").await.unwrap());
+        info!("Bound socket: {}", socket.local_addr().unwrap());
+        info!("Trying to connect to: {}", url);
+        let _ = socket.connect(url).await;
```

Comment:
Don't we want to return and report an error in this case, rather than silently fail?

Reply:
Yeah, I'm just not sure if we want to do it all and return a `Result` from the `new` method in the long term, or separate the creation of the Client and some `init` method that would deal with all the possible issues.